### ISSUE TITLE
Setting cell value to empty string has different results on Excel for Mac and Excel Online w.r.t. ISBLANK formula

### ISSUE THREAD

**Issue by Rinzwind** [OfficeDev/office-js#1943]
The Script Lab snippet below has different results on Excel Online and Excel for Mac. The ‘run’ function of the snippet is:

```
const sheet = context.workbook.worksheets.getActiveWorksheet();
const rangeData = sheet.getRange("A1:A2");
const rangeFormula = sheet.getRange("B1:B2");
(rangeData.numberFormat as unknown) = "@";
rangeData.values = [[""], ["abc"]];
(rangeFormula.formulasR1C1 as unknown) = "=ISBLANK(RC[-1])";
await context.sync();
```

On Excel for Mac (v16.50 21061301, macOS 10.15.7) the result is:

<img width="159" alt="1" src="https://user-images.githubusercontent.com/1611248/123076582-92607d00-d419-11eb-9b9a-a183ce42297e.png">

On Excel Online (using Chrome 91.0.4472.114) the result is:

<img width="179" alt="2" src="https://user-images.githubusercontent.com/1611248/123076631-a0160280-d419-11eb-931d-037f1f4d6baa.png">

The screenshots are in Dutch: WAAR = TRUE, ONWAAR = FALSE (my apologies, I tried changing the language to English but had some trouble doing so). The difference between the two screenshots is in cell B1: ONWAAR (FALSE) on the first screenshot, WAAR (TRUE) on the second screenshot.

Why are these results different? Shouldn’t both Excel platforms have the same behavior?

The snippet is based on the one I posted in issue #931, but I tried to simplify it to better clarify the issue.

**Complete Script Lab Snippet:**

```
name: Test
description: ''
host: EXCEL
api_set: {}
script:
  content: |
    $("#run").click(() => tryCatch(run));

    async function run() {
      await Excel.run(async (context) => {
        const sheet = context.workbook.worksheets.getActiveWorksheet();
        const rangeData = sheet.getRange("A1:A2");
        const rangeFormula = sheet.getRange("B1:B2");
        (rangeData.numberFormat as unknown) = "@";
        rangeData.values = [[""], ["abc"]];
        (rangeFormula.formulasR1C1 as unknown) = "=ISBLANK(RC[-1])";
        await context.sync();
      });
    }

    /** Default helper for invoking an action and handling errors. */
    async function tryCatch(callback) {
      try {
        await callback();
      } catch (error) {
        // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
        console.error(error);
      }
    }
  language: typescript
template:
  content: |-
    <section class="samples ms-font-m">
      <button id="run" class="ms-Button">
        <span class="ms-Button-label">Run</span>
      </button>
    </section>
  language: html
style:
  content: |-
    section.samples {
        margin-top: 20px;
    }

    section.samples .ms-Button, section.setup .ms-Button {
        display: block;
        margin-bottom: 5px;
        margin-left: 20px;
        min-width: 80px;
    }
  language: css
libraries: |
  https://appsforoffice.microsoft.com/lib/1/hosted/office.js
  @types/office-js

  office-ui-fabric-js@1.4.0/dist/css/fabric.min.css
  office-ui-fabric-js@1.4.0/dist/css/fabric.components.min.css

  core-js@2.4.1/client/core.min.js
  @types/core-js

  jquery@3.1.1
  @types/jquery@3.3.1

```

**Comment by keyur32** [comment]
@lumine2008  can we take a look? thx

**Comment by jipinshi** [comment]
@Rinzwind, Thanks reaching us. I can repro the issue now. It looks like ISEMPTY function behavior is not same between online and win32/mac, online will treat "" as true while win32/mac will treat it as false. So one way to unblock you is add additional judgement like =IF(LEN(A1)=0,"TRUE","FALSE")

**Comment by Rinzwind** [comment]
@jipinshi: thanks for the suggestion, and for confirming the issue can be reproduced. I took another look at my snippet, if I write the ‘run’ function as:

```typescript
const sheet = context.workbook.worksheets.getActiveWorksheet();
const rangeData = sheet.getRange("A1:A2");
const rangeFormula = sheet.getRange("B1:B2");
(rangeData.numberFormat as unknown) = "@";
// Replaced:
//   rangeData.values = [[""], ["abc"]];
// By:
rangeData.getCell(0, 0).values = [[""]];
rangeData.getCell(1, 0).values = [["abc"]];
// --
(rangeFormula.formulasR1C1 as unknown) = "=ISBLANK(RC[-1])";
await context.sync();
```

Then the result on Excel for Mac is:

<img width="161" alt="3" src="https://user-images.githubusercontent.com/1611248/123221914-b7172c00-d4cf-11eb-823a-2be7cdc88ab9.png">

Note that cell B1 is now WAAR (TRUE).

So: the original single assignment (`rangeData.values = [[""], ["abc"]];`) leads to cell B1 being ONWAAR (FALSE), while the replacement with two assignments (`rangeData.getCell(0, 0).values = [[""]]; rangeData.getCell(1, 0).values = [["abc"]];`) leads to cell B1 being WAAR (TRUE). It looks like this is an inconsistency in the behavior of the Excel JavaScript API on Mac?

**Comment by jipinshi** [comment]
@Rinzwind , so great to hear you find another good way to unblock you. Yes, it is one bug about the consistence, 5218458 was created to do the track internally.

**Comment by gingerjia** [comment]
@Rinzwind, when we try to investigate details about the root cause, we found the issue not repro now. My excel build is:
![image](https://user-images.githubusercontent.com/62282034/128676337-cdb02681-8dbd-4de6-8470-ccc758bcf903.png)
 
Can you help to double confirm at your side? If still repro, please let us know. Thanks very much!

**Comment by Rinzwind** [comment]
@gingerjia: I can still reproduce the issue. Using the “Complete Script Lab Snippet” (from my [first comment](https://github.com/OfficeDev/office-js/issues/1943#issue-928078804)), I get:

On Excel for Mac (v16.51 21071101, macOS 10.15.7):

<img width="160" alt="Excel for Mac" src="https://user-images.githubusercontent.com/1611248/128684855-60309042-928c-4eac-931f-444a3972a0c2.png">

On Excel Online (using Chrome 92.0.4515.131):

<img width="171" alt="Excel Online" src="https://user-images.githubusercontent.com/1611248/128684872-0a6f74b2-774a-4709-b7fb-72717385b15f.png">

**Comment by gingerjia** [comment]
@Rinzwind, thanks very much for helping confirm. We will do the further investigation and update to you soon when have any progress. Thanks again!

**Comment by gingerjia** [comment]
@Rinzwind sorry for late response. We can repro the issue at 16.51.
And we did a fix for it and the fix has been deployed to 16.52  version. 
Could you help to upgrade your build version to 16.52 and have a try when you are convenient? 

If any other issues please let us know! Thanks!

**Comment by Rinzwind** [comment]
@gingerjia: using the “Complete Script Lab Snippet” again, I get:

On Excel for Mac (16.52 21080801, macOS 10.15.7):

<img width="162" src="https://user-images.githubusercontent.com/1611248/129852205-5b5a9084-3599-4b41-bdf0-5b58a5237fcd.png">

So I can confirm it has been fixed. I will close this issue, thanks!
